### PR TITLE
build: update to latest sass version

### PR DESCRIPTION
**Pull request by crisbeto**
Bumps to the latest version of Sass and fixes up our custom patch so that it accounts for some changes in the published files.

### CODE REVIEW

File: tools/postinstall/sass_worker_async.patch [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
    debug('Starting Sass compiler persistent worker...');
--  runWorkerLoop(args => sass.run_(args));
+-  runWorkerLoop(args => sass.cli_pkg_main_0_(args));
 +  runWorkerLoop(args => {
 +    return new Promise(resolve => {
-+      sass.run_(args)['then$1$2$onError'](() => resolve(true), () => resolve(false));
++      const result = sass.cli_pkg_main_0_(args);
++      const originalThen = result.then$1$2$onError;
++      let dynamicType;
++      result.then$1$2$onError = (receiver, done, error, type) => dynamicType = type;
```
Comment (crisbeto) [inline tools/postinstall/sass_worker_async.patch:15]: This is a very janky implementation, but the problem it's trying to solve is that `then$1$2$onError` requires an internal Dart type object to be passed in which we don't have access to. I hacked around it by monkey-patching the function, saving a reference and then passing it back in.

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by mmalerba** [comment]
@crisbeto needs rebase

**Comment by crisbeto** [comment]
Rebased.

**Pull request merged**